### ISSUE TITLE
Select query with multiple joins won't work

### ISSUE THREAD

**Issue by ericchuawc** [Altinity/clickhouse-mysql-data-reader#181]
I need some help.

I have managed to migrate some of MySQL 5.1 tables to Clickhouse.

Basic select query works fine with 1 inner join.

However I noticed if I have more than 1 join e.g.

select 
from table1
inner join table2
on table1.a = table2.a
inner join table3
... (more joins left join or inner joins)

This won't work in ClickHouse (it works in MySQL)

Do you have any recommendations how to rewrite this query inside ClickHouse? Otherwise what is the best approach to do this?

Thanks.

**Comment by sv3t0sl4v** [comment]
Yes. With sub-queries with good WHERE and/or HAVING statements most probably.
